workers/ohsome_quality_analyst/ohsome/client.py

```diff
@@ -20,45 +20,38 @@ async def query(
     ratio: bool = False,
 ) -> Optional[Dict]:
     """Query ohsome API endpoint with filter."""
-    if endpoint is not None:
-        url = OHSOME_API + endpoint
-    else:
-        url = OHSOME_API + layer.endpoint
-
-    if ratio:
-        url = url + "/ratio"
-        data = {"bpolys": bpolys, "filter": layer.filter, "filter2": layer.ratio_filter}
-    else:
-        data = {"bpolys": bpolys, "filter": layer.filter}
-
-    if time is not None:
-        data["time"] = time
-
+    url = build_url(layer, endpoint, ratio)
+    data = build_data_dict(layer, bpolys, time, ratio)
     logging.info("Query ohsome API.")
-    logging.debug("Query data: " + json.dumps(data, indent=4, sort_keys=True))
+    logging.debug("Query URL: " + url)
+    logging.debug("Query filter: " + layer.filter)
+    logging.debug("Query data: " + json.dumps(data))
+    return await query_ohsome_api(url, data)
+
 
-    # set custom timeout as ohsome API takes a long time to send an answer
-    timeout = httpx.Timeout(5, read=600)
+async def query_ohsome_api(url: str, data: dict):
+    # set custom timeout as ohsome API takes a long time (10 minutes) to send an answer
+    timeout = httpx.Timeout(5, read=660)
     async with httpx.AsyncClient(timeout=timeout) as client:
         resp = await client.post(url, data=data)
 
-    logging.info("Query ohsome API.")
-    logging.debug("Query URL: " + url)
-    logging.debug("Query Filter: " + layer.filter)
-    if resp.status_code == 200:
-        try:
-            logging.info("Ohsome query successful!")
-            logging.debug(
-                "Query response: " + json.dumps(resp.json(), indent=4, sort_keys=True)
-            )
-            return geojson.loads(resp.content)
-        except JSONDecodeError:
-            # ohsome API can return broken GeoJSON during streaming of response
-            logging.warning("Ohsome query failed!")
-            return None
-    elif resp.status_code == 404:
-        logging.warning("Query failed!")
-        return None
+    try:
+        resp.raise_for_status()  # Raise for response status codes 4xx and 5xx
+    except httpx.HTTPStatusError:
+        logging.error("Query ohsome API failed!")
+        raise
```

Comment:
Can we provide more context then stating that quering the ohsome API failed since response status code could be any of 4xx and 5xx? Should we state more clearly that the reponse code has bee returend from the ohsome API?

Reply:
Does commit b390b848792f0df4aeb29d280e1ea197247ed32fb adresses the problem you raised here?